newIDE/app/src/ProjectsStorage/LocalFileStorageProvider/LocalFileResourceMover.js

```diff
@@ -54,49 +96,66 @@ export const moveUrlResourcesToLocalFiles = async ({
       const resource = resourcesManager.getResource(resourceName);
 
       const url = resource.getFile();
-      let filenameWithExtension;
-      if (isProductAuthorizedResourceUrl(url)) {
-        // Resource is a private asset.
-        filenameWithExtension = extractFilenameWithExtensionFromProductAuthorizedUrl(
-          url
-        );
-      } else if (isPublicAssetResourceUrl(url)) {
-        // Resource is a public asset.
-        filenameWithExtension = extractFilenameWithExtensionFromPublicAssetResourceUrl(
-          url
-        );
-      } else {
-        // Resource is a project resource or a generic url.
-        filenameWithExtension = extractFilenameFromProjectResourceUrl(url);
-      }
-      const extension = path.extname(filenameWithExtension);
-      const filenameWithoutExtension = path.basename(
-        filenameWithExtension,
-        extension
-      );
-      const name = newNameGenerator(filenameWithoutExtension, name => {
-        const tentativePath = path.join(baseAssetsPath, name) + extension;
-        return (
-          fs.existsSync(tentativePath) || downloadedFilePaths.has(tentativePath)
-        );
-      });
-      const downloadedFilePath = path.join(baseAssetsPath, name) + extension;
-      downloadedFilePaths.add(downloadedFilePath);
+      if (isBlobURL(url)) {
+        try {
+          const {
+            localFilePath,
+            extension,
+          } = parseLocalFilePathOrExtensionFromMetadata(resource);
+          const downloadedFilePath = localFilePath
+            ? path.resolve(projectPath, localFilePath)
+            : generateUnusedFilepath(
+                baseAssetsPath,
+                downloadedFilePaths,
+                resource.getName() + (extension || '')
+              );
 
-      try {
-        await retryIfFailed({ times: 2 }, async () => {
           await fs.ensureDir(baseAssetsPath);
-          await ipcRenderer.invoke(
-            'local-file-download',
-            url,
-            downloadedFilePath
-          );
+          await downloadBlobToLocalFile(url, downloadedFilePath);
           resource.setFile(
             path.relative(projectPath, downloadedFilePath).replace(/\\/g, '/')
           );
-        });
-      } catch (error) {
-        erroredResources.push({ resourceName, error });
+        } catch (error) {
+          erroredResources.push({ resourceName, error });
+        }
+      } else {
+        let filename;
```

Comment:
should we keep filenameWithExtension to be really specific?

Reply:
I would go for filename because the extension is part of a filename (also called "basename" sometimes). So I would rather call "filenameWithoutExtension" the special case when the extension is removed?

Reply:
Alright if we go with that, then I would update everywhere in the code where it's the case (ex: CloudResourceFetcher)